mirdata/datasets/saraga.py

```diff
@@ -0,0 +1,621 @@
+                self._track_metadata = {
+                    'raaga': None,
+                    'form': None,
+                    'title': None,
+                    'work': None,
+                    'length': None,
+                    'taala': None,
+                    'album_artists': None,
+                    'mbid': None,
+                    'artists': None,
+                    'concert': None
+                }
+
+            self.raaga = self._track_metadata['raaga']
+            self.form = self._track_metadata['form']
+            self.title = self._track_metadata['title']
+            self.work = self._track_metadata['work']
+            self.taala = self._track_metadata['taala']
+            self.album_artists = self._track_metadata['album_artists']
+            self.mbid = self._track_metadata['mbid']
+            self.artists = self._track_metadata['artists']
+            self.concert = self._track_metadata['concert']
+
+        # HINDUSTANI MUSIC TRACKS
+        if self.iam_style == 'hindustani':
+            metadata = DATA.metadata(self.metadata_path)
+            if metadata is not None and track_id == metadata['track_id']:
+                self._track_metadata = metadata
+            else:
+                # annotations with missing metadata
+                self._track_metadata = {
+                    'title': None,
+                    'raags': None,
+                    'length': None,
+                    'album_artists': None,
+                    'forms': None,
+                    'mbid': None,
+                    'artists': None,
+                    'release': None,
+                    'works': None,
+                    'taals': None,
+                    'layas': None
+                }
+
+            self.title = self._track_metadata['title']
+            self.raags = self._track_metadata['raags']
+            self.album_artists = self._track_metadata['album_artists']
+            self.forms = self._track_metadata['forms']
+            self.mbid = self._track_metadata['mbid']
+            self.artists = self._track_metadata['artists']
+            self.release = self._track_metadata['release']
+            self.works = self._track_metadata['works']
+            self.taals = self._track_metadata['taals']
+            self.layas = self._track_metadata['layas']
+
+    @utils.cached_property
+    def tonic(self):
+        """Float: tonic annotation"""
+        return load_tonic(self.ctonic_path)
+
+    @utils.cached_property
+    def pitch(self):
+        """F0Data: pitch annotation"""
+        return load_pitch(self.pitch_path)
+
+    @utils.cached_property
+    def pitch_vocal(self):
+        """F0Data: pitch vocal annotations"""
+        return load_pitch(self.pitch_vocal_path)
+
+    @utils.cached_property
+    def tempo(self):
+        """Dict: tempo annotations"""
+        if self.iam_style == 'carnatic':
+            return load_tempo_carnatic(self.tempo_path)
```

Comment:
nit: can you have a single `load_tempo(tempo_path, iam_stle)`?

As much as possible, we're trying to have e.g. `self.tempo` to return roughly the same thing as `load_tempo` without too much extra logic.

Reply:
Yes! Fixed now.